agents/rust/mediator/mediation/src/routes/pickup.rs

```diff
@@ -4,66 +4,81 @@ use std::sync::Arc;
 
 use axum::{extract::State, http::StatusCode, Json};
 use log::info;
-
-use crate::{
-    didcomm_types::{
-        pickup_delivery_message_structs::*, PickupDeliveryReqMsg, PickupMsgEnum, PickupStatusMsg,
-        PickupStatusReqMsg, ProblemReportMsg,
+use messages::{
+    decorators::attachment::{Attachment, AttachmentData, AttachmentType},
+    msg_fields::protocols::pickup::{
+        Delivery, DeliveryContent, DeliveryRequestContent, Pickup, Status, StatusContent,
+        StatusDecorators, StatusRequestContent,
     },
-    storage::MediatorPersistence,
 };
+use uuid::Uuid;
+
+use crate::storage::MediatorPersistence;
 
-pub async fn handle_pickup<T: MediatorPersistence>(
+pub async fn handle_pickup_authenticated<T: MediatorPersistence>(
     State(storage): State<Arc<T>>,
-    Json(pickup_message): Json<PickupMsgEnum>,
-) -> (StatusCode, Json<PickupMsgEnum>) {
+    Json(pickup_message): Json<Pickup>,
+    auth_pubkey: &str,
+) -> (StatusCode, Json<Pickup>) {
```

Comment:
I could imagine that this function could return custom `Result` type (using custom `MediatorError` type).
In that model you would typically return `Ok(response)` or `Error(<some_err_info_struct/enum>)`